### PR TITLE
Send Redis reports to a secondary Redis

### PR DESCRIPTION
**Pull request by uaalto**
This activates basic and anonymous, traffic reporting to a secondary Redis.

### CODE REVIEW

File: salt/http_proxy/config.ini [hunk 1]
```diff
@@ -13,6 +13,8 @@ logglytoken = {{ pillar['loggly_token'] }}
 pprofaddr = localhost:4000
 proxied-sites-sample-percentage = 0.0001
 proxied-sites-tracking-id = UA-21815217-16
+redis = pub-redis-18876.us-east-1-3.1.ec2.garantiadata.com:18876
```
Comment (aranhoide) [inline salt/http_proxy/config.ini]: Sorry, I told you in slack that this was OK for testing, but it is not really, especially if such testing is going to happen in the production deployment:

http://antirez.com/news/96

(tl;dr: an unprotected redis instance can give you access to the host as the `redis` user.)

Also, if this is deployed then random people can read, modify, or flush this database, causing us unnecessary confusion at best.

tl;dr If this is going to be deployed to production, even if only for testing, it needs a password.  If it's fine for you, let me know and I'll send you a PR into this one to add a password (which we'll also need to set it in the redis admin page before deploying).

**Comment by aranhoide** [comment]
I trust that you've tested `enablereports` to be the right config key to use.  Barring the security issue (and sorry for not having brought it up in slack, but I didn't know you meant to test this in master), this LGTM.

If your plan is not to deploy this to the production setup yet then it needs to live in a separate branch.  Master should always be ready to deploy to production.

**Comment by uaalto** [comment]
Yes, let's add the password. I'll do it.

**Comment by uaalto** [comment]
I just pushed the necessary changes for the proxy. This is the PR: https://github.com/getlantern/http-proxy-lantern/pull/59
I've assigned xiam, but feel free to take over if you like, so we get this to production ASAP.

Now I'll tackle the Salt side of things, I might need some hand holding for this...

**Comment by uaalto** [comment]
hey @aranhoide, the support for authentication should be complete in both the proxy and the Salt configuration.

**Comment by aranhoide** [comment]
LGTM.

**Comment by aranhoide** [comment]
@uaalto: I'll deploy this as soon as I've reviewed the rest of Ox's DOS changes, or when I see you in Slack, whatever happens first.

**Comment by uaalto** [comment]
Awesome! Thank you
